x-pack/plugins/endpoint/public/applications/endpoint/view/alerts/index.tsx

```diff
@@ -4,41 +4,94 @@
+import { useAlertListSelector } from './hooks/use_alerts_selector';
 
 export const AlertIndex = memo(() => {
-  usePageId('alertsPage');
+  const history = useHistory();
 
-  const columns: Array<{ id: string }> = useMemo(() => {
+  const columns: EuiDataGridColumn[] = useMemo(() => {
     return [
-      { id: 'alert_type' },
-      { id: 'event_type' },
-      { id: 'os' },
-      { id: 'ip_address' },
-      { id: 'host_name' },
-      { id: 'timestamp' },
-      { id: 'archived' },
-      { id: 'malware_score' },
+      {
+        id: 'alert_type',
+        display: i18n.translate('xpack.endpoint.application.endpoint.alerts.alertType', {
+          defaultMessage: 'Alert Type',
+        }),
+      },
+      {
+        id: 'event_type',
+        display: i18n.translate('xpack.endpoint.application.endpoint.alerts.eventType', {
+          defaultMessage: 'Event Type',
+        }),
+      },
+      {
+        id: 'os',
+        display: i18n.translate('xpack.endpoint.application.endpoint.alerts.os', {
+          defaultMessage: 'OS',
+        }),
+      },
+      {
+        id: 'ip_address',
+        display: i18n.translate('xpack.endpoint.application.endpoint.alerts.ipAddress', {
+          defaultMessage: 'IP Address',
+        }),
+      },
+      {
+        id: 'host_name',
+        display: i18n.translate('xpack.endpoint.application.endpoint.alerts.hostName', {
+          defaultMessage: 'Host Name',
+        }),
+      },
+      {
+        id: 'timestamp',
+        display: i18n.translate('xpack.endpoint.application.endpoint.alerts.timestamp', {
+          defaultMessage: 'Timestamp',
+        }),
+      },
+      {
+        id: 'archived',
+        display: i18n.translate('xpack.endpoint.application.endpoint.alerts.archived', {
+          defaultMessage: 'Archived',
+        }),
+      },
+      {
+        id: 'malware_score',
+        display: i18n.translate('xpack.endpoint.application.endpoint.alerts.malwareScore', {
+          defaultMessage: 'Malware Score',
+        }),
+      },
     ];
   }, []);
 
-  const [visibleColumns, setVisibleColumns] = useState(() => columns.map(({ id }) => id));
+  const { pageIndex, pageSize, total } = useAlertListSelector(selectors.alertListPagination);
+  const urlFromNewPageSizeParam = useAlertListSelector(selectors.urlFromNewPageSizeParam);
+  const urlFromNewPageIndexParam = useAlertListSelector(selectors.urlFromNewPageIndexParam);
+  const json = useAlertListSelector(selectors.alertListData);
+
+  const onChangeItemsPerPage = useCallback(
+    newPageSize => history.push(urlFromNewPageSizeParam(newPageSize)),
+    [history, urlFromNewPageSizeParam]
+  );
+
+  const onChangePage = useCallback(
+    newPageIndex => history.push(urlFromNewPageIndexParam(newPageIndex)),
+    [history, urlFromNewPageIndexParam]
+  );
 
-  const json = useSelector(selectors.alertListData);
+  const [visibleColumns, setVisibleColumns] = useState(() => columns.map(({ id }) => id));
```

Comment:
Why use of local state and not redux store?

Reply:
i dont know if we had a particular reason for not putting it in store beyond we didn't consider it part of the feature and it was just a side effect already built into the eui DataGrid component